### ISSUE TITLE
Your main page at http://ubjson.org/ has been hacked.

### ISSUE THREAD

**Issue by badgermole** [ubjson/universal-binary-json#108]
There are many inserted links and text for viagra, etc.  Please bring it down and fix asap.
![image](https://user-images.githubusercontent.com/33312370/70938720-98db9080-1ffb-11ea-9b4c-1ae2b106726b.png)

**Comment by rkalla** [comment]
Chrome / Windows 10 - I don't see this (no plugins blocking).

Can you give me steps to reproduce?

**Comment by AnyCPU** [comment]
@rkalla Confirmed, there are a lot of external link. The main page is doorway at the moment.
@badgermole Thanks!

**Comment by rkalla** [comment]
Damnit - looks like it's serving a different version to Chrome. I'll see if
I can clean it up. Thank you!

**Comment by fangq** [comment]
in addition to the defaced web pages, there is also a list of warnings from the server php 

```
Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-adspace.php on line 204

Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-blogauthor.php on line 244

Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-embed.php on line 229

Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-flickr.php on line 236

Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-subscribe.php on line 175

Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-tabs.php on line 231

Deprecated: Function create_function() is deprecated in /var/www/wp-content/themes/canvas/includes/widgets/widget-woo-component.php on line 238
```

**Comment by fangq** [comment]
as of today, the spam appears to have been removed, but the php warnings still appear at the beginning and the bottom of the page.

**Comment by Enichan** [comment]
Still showing spam for me on Win10/Firefox and Google still warns that the page has been hacked
![image](https://user-images.githubusercontent.com/3856285/72981229-ed63ef80-3da1-11ea-829b-43bbbec15b6e.png)

**Comment by nyue** [comment]
I am still seeing PHP server deprecation errors

**Comment by Enichan** [comment]
Yeah, spam is still there too. With the page having been in this state for this long I can't in good conscience recommend any of my peers to take a look at UBJSON, which is sad since I think it's probably the most elegantly designed binary JSON format and I spent quite some time implementing it myself. :/

**Comment by fangq** [comment]
For those who are interested in continue developing UBJSON format, I am going to start a derived specification (named **Binary JData Specification**) as part of my OpenJData Project (http://openjdata.org, http://openjdata.org/wiki/) and JData Specification (https://github.com/fangq/jdata)

The current binary JData specification already differs from UBJSON by adding a [non-compatible ND-array construct](https://github.com/fangq/jdata/blob/master/JData_specification.md#binary-jdata-storage-grammar), which I previously discussed in these threads: https://github.com/ubjson/universal-binary-json/issues/61#issuecomment-492445324 and https://github.com/ubjson/universal-binary-json/issues/101

I am working on updating all UBJSON parsers to support this ND array container. Another immediate extension I would like to make is to add the missing `uint16`, `uint32` and `uint64` with a possible `float16` (or half-precision) markers, so that I can unambigiously map my existing JData array types to this binary format

https://github.com/fangq/jdata/blob/master/JData_specification.md#annotated-storage-of-n-d-arrays

Other than these extensions, I am also interested in finishing the design of the template support. I started a thread of discussion in the messagepack project

https://github.com/msgpack/msgpack/pull/267

but the template support is not currently a priority. 

I will set up a MarkUp based specification draft and announce the link here. I encourage everyone who loves and had contributed to this specification to continue your support in the new project.

**Comment by nyue** [comment]
The unsigned counterpart completeness is very welcome, as does float16 (I do some work with OpenEXR)

**Comment by fangq** [comment]
Here is the repo link: https://github.com/OpenJData/bjdata

I started from a MarkDown [document](https://github.com/Iotic-Labs/py-ubjson/blob/dev-contrib/UBJSON-Specification.md) found in the [py-ubjson repo](https://github.com/Iotic-Labs/py-ubjson), which is aimed to summarize  the Draft 12 of the UBJSON specification.

From that starting point, I added the extended syntax, https://github.com/OpenJData/bjdata/commit/95544dc8a999cec1d019e22f24fccc70bbdf5073

this includes the uint16 (u), uint32 (m), uint64 (M), half (h) markers and optimized ND packed array support https://github.com/OpenJData/bjdata/commit/95544dc8a999cec1d019e22f24fccc70bbdf5073#diff-d94a316d99f805938f1320db036a84a5R380

If you are interested, please check the rest of the specification and see if any of those are not in agreement with Draft 12 of UBJSON. I will continue polishing the description, adding new examples and review new feature requests in the bjdata repository

https://github.com/OpenJData/bjdata/issues

**Comment by fangq** [comment]
For those who are interested, I just released the [**Draft 1** of the Binary JData spec](https://github.com/OpenJData/bjdata/tree/Draft_1).

I would like you to help review the formatted specification at
https://github.com/OpenJData/bjdata/blob/Draft_1/Binary_JData_Specification.md

and share your comments and suggestions on [this page](https://github.com/OpenJData/bjdata/commit/2f352210f928af9efd89f47c845d8e68c5b9bbc2)

In short, 
```
Binary JData Draft 1 = 
          UBJSON Draft 12 
        + uint16/uint32/uint64/float16 markers 
        + ND-array optimized header 
        - NaN->null rule. 
```

if you see any thing that deserves further clarifications, please feel free to create an issue or submit a PR. 

Currently, I have included 3 BJData Draft1 compliant parsers, linked as submodules in the [bjdata/lib](https://github.com/fangq/bjdata/tree/master/lib) folder
- **JSONLab** for MATLAB (I wrote it), 
- a modified **py-ubjson** for Python (I submitted a [pull request](https://github.com/Iotic-Labs/py-ubjson/pull/14)), and 
- a modified **ubj** (written by @Steve132) for C/C++ (I submitted a [pull request](https://github.com/Steve132/ubj/pull/8) for the ND array header, still need to patch for the new data types)

Your comments are welcome.

**Comment by Steve132** [comment]
Unfortunately, I don't have control over the ubjson.org domain.  I can search to find out who does.

Is there any way we can standardize on an array standard?  If the ND array optimized header is different it seems like its better not to duplicate work.    I'd be down to incorporate uint16 uint32 uint64 and float16 into UBJSON and standardize it so that we can have a single standard.

**Comment by fangq** [comment]
hi @Steve132, I agree that we all prefer not to have duplicated works. I am absolutely happy if someone is willing to back-port these features to UBJSON so that our parsers are 100% compatible. 

On the other hand, aside from the crippled website, the latest update to the UBJSON repo is over 2 years ago, making many people who had invested time in this project feeling disappointed because no one is maintaining the project.

In the short term, I do hope to at least maintain a parallel spec specifically for my OpenJData project because it moves at a much faster pace (I am preparing on a resubmission of a related grant proposal for the NIH). Having my fork will allow me to have flexibility to prioritize features that is essential for my other related specifications and the parsers that I am maintaining.

**Comment by Steve132** [comment]
I accept pull requests and maintenance requests.  I too am disappointed
that not a lot of progress has been made.

**Comment by Steve132** [comment]
@rkalla  Do you have control over ubjson.org?  Thebuzzmedia seems to just be gone.

**Comment by Steve132** [comment]
I vote we point the domain to our github pages here.

**Comment by edgar-bonet** [comment]
@Steve132 wrote:
> I accept pull requests and maintenance requests.

Will you have a chance to look at those that are currently open?

**Comment by Steve132** [comment]
I've started looking through them all yesterday.   They are broadly in 3
categories:  A) Bugs in the current spec (e.g., the website being down, the
spec versions not being labeled, the spec being in HTML vs markdown, the
spec underspecifying endian-ness or correct behavior or not being a formal
RFC).   All of these in category A should get fixed trivially and I'm
starting to work on them.

B) is small new features.   things like new data types, endian-ness, new
flags, whether or not to change the order of the parser, streaming headers,
stuff like that.

C) is nd-array or templating or typespec.   There are about 10 of these
suggestions spread out over multiple issues.  I'm going to organize all of
these into a single document in terms of trying to move towards a version
1.0 that includes at least one solution.  (And no, I'm not going to
railroad my '@' proposal just because it's mine.  I spent about 4 hours
last night reading literally everything and I'm going to organize it and
discuss before moving towards a solution with the community).

**Comment by kag0** [comment]
https://github.com/ubjson/universal-binary-json/pull/110
